### ISSUE TITLE
Combine/reduce some common code libraries.

### ISSUE THREAD

**Issue by samreid** [phetsims/chipper#1346]
From an idea by @pixelzoom in https://github.com/phetsims/chipper/issues/1345#issuecomment-1285981809

> To make it feasible/scalable to create/test/maintain stable versions of common libraries, I think PhET needs to reduce the number of common libraries, by combining them into a smaller number of libraries (but NOT a mon repo). And I think that consolidation should be one of the first steps -- certainly before creating 1.0 branches of existing repos, as is proposed above.

I do not think this is necessary--it doesn't matter too much whether we have N or N+M libraries (as long as N>1), we will still need multi-repo tooling.  But I agree some ones could be combined.  I'll paste the common repos here and we can brainstorm what should be combined.  This list from build.json:

* assert
* axon
* babel
* brand
* chipper
* dot
* joist
* kite
* perennial-alias
* phet-core
* phetcommon
* phetmarks
* query-string-machine
* scenery
* scenery-phet
* sherpa
* sun
* tambo
* tandem
* twixt
* utterance-queue

I'll go first: not knowing too much about it, but I feel utterance-queue could be integrated into scenery.  Maybe @jessegreenberg or @zepumph can comment on why it was created separate?

Also, assert could be streamlined into phet-core.

I'm not sure why phetmarks is listed as a dependency.  Yes, it's nice to get its shas, but it's not part of a build.

phetcommon should be eliminated.

Assigning to @pixelzoom to get more feedback and ideas.

**Comment by pixelzoom** [comment]
> I do not think this is necessary--it doesn't matter too much whether we have N or N+M libraries (as long as N>1), we will still need multi-repo tooling. 

Agreed that it's not technically necessary for tooling.  It is mostly-definitely necessary in order to make the cost feasible. PhET cannot afford release processes for 21 common-code libraries.  If there are that many libraries, we will never get them releases in a reasonable amount of time, with any kind of regularity, or at a cost that the project can bear.

**Comment by samreid** [comment]
Any comments on the combos/reductions proposed above?

**Comment by pixelzoom** [comment]
I'm not clear on what you're asking. Are you asking how I would combine the existing common-code repos?

**Comment by samreid** [comment]
Here are some proposals from the top comment.  Do they sound good to you? Do you want to add to them?

* utterance-queue could be integrated into scenery.
* assert could be streamlined into phet-core
* eliminate phetcommon
* phetmarks may not need to be maintained in the same way as standard dependenices

**Comment by pixelzoom** [comment]
Off the top of my head....

phet-core (or another name, if you prefer) becomes most of the libraries that are required (not optional) to create PhET sims, with the exception of libraries that PhET would like to be useful outside of PhET sims (scenery, dot, kit, sun,...)  So the consolidation is something like:
* phet-core = axon + brand + joist + phet-core + phetcommon + query-string-machine + scenery-phet + tambo + tandem

For any consolidation, we'll need to discuss repo structure.  But most of the above could simply be subdirectories of phet-core/js/.

As you said above, utterance-queue seems appropriate to integrate into scenery. So:
* scenery = scenery + utterance-queue

utterance-queue/ would probably be another subdirectory of scenery/js/.

These libraries are unchanged:
* babel
* chipper
* dot
* kite
* perennial
* sherpa
* sun
* twixt

**Comment by pixelzoom** [comment]
I also think that private libraries (PhET-iO) should be included in this plan.  There's value in versioning them too, and it's perhaps even more important for a commerical product like PhET-iO.

**Comment by samreid** [comment]
I can see the value in combining brand into joist.  I feel assert could move into phet-core.

To me, axon seems like a standalone library like we want dot or kite to be, competing with the data layers defined by Backbone, Ember or Knockout.

query-string-machine seems like a useful standalone library that may be good to keep independent.

But I also feel that many of the cons for monorepos identified in https://github.com/phetsims/chipper/issues/1242 apply (to a lesser extent) to making a smaller monorepo.

**Comment by samreid** [comment]
> phet-core = axon + brand + joist + phet-core + phetcommon + query-string-machine + scenery-phet + tambo + tandem

If we merge axon, brand, joist, tambo, etc into phet-core, then the GitHub issues for that repo are going to be very messy--we would want labels to keep track of which is which.  Like a phet-core label "type: axon". Kind of like we had in Unfuddle which didn't work very well.  It seems preferable to keep all of the tambo issues in the tambo repo.

**Comment by samreid** [comment]
I wrote some other comments in https://github.com/phetsims/chipper/issues/1345#issuecomment-1289891785 about combining repos:

> In my opinion, merging some repos does not fundamentally change the nature of the problem. It does not give us atomic commits. It does not change the scripting/tooling we would need. Yes, we incrementally decrease the number of repos/commits/merges/changelogs/dependencies, but we lose modularity and modular GitHub issue tracking and have to pay an up-front merge cost.
